lib/utils/utils.dart

```diff
@@ -58,6 +58,15 @@ void checkAndExecuteOrNot(String macAddress, String ipAddress, BuildContext cont
   }
 }
 
+void checkAndExecuteOrNotNE(String macAddress, String ipAddress) async {
```

Comment:
At the moment this entire method could be removed, as nothing happens if there's no internet anyway.
I'd move it to some controller class, first check if there's internet connection (if no then display snackbar) and then call sendMagicPacket.

Reply:
I agree, but  I want to keep this solution at the moment.